### PR TITLE
feat: add remove building unit domain

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/RemoveBuildingUnitLambdaHandler.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            {
+                // Idempotent: Do Nothing return last etag
+            }
+
+            var lastHash = await GetHash(
+                request.BuildingPersistentLocalId,
+                new BuildingUnitPersistentLocalId(request.BuildingUnitPersistentLocalId),
+                cancellationToken);
+
+            return new ETagResponse(string.Format(DetailUrlFormat, request.BuildingUnitPersistentLocalId), lastHash);
```
Comment (emalfroy) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/RemoveBuildingUnitLambdaHandler.cs:55]: @ArneD should we include lastHash in etag response or pass empty string?

Comment (ArneD) [reply src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/RemoveBuildingUnitLambdaHandler.cs:55]: Last hash is good (consistent with address).

File: src/BuildingRegistry/Building/BuildingUnit.cs [hunk 2]
```diff
@@ -248,6 +247,18 @@ public void CorrectRetiredBuildingUnit(BuildingGeometry buildingGeometry)
             return correctedBuildingUnitPosition;
         }
 
+        public void Remove()
+        {
+            GuardCommonUnit();
```
Comment (ArneD) [inline src/BuildingRegistry/Building/BuildingUnit.cs:252]: Will ask business about this.

**Review by ArneD — CHANGES_REQUESTED** [review]

**Review by ArneD — APPROVED** [review]

**Pull request merged**